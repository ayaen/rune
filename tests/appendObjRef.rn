//  Copyright 2023 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class Foo(self, name) {
  self.name = name

  final(self) {
    println "Destroying ", self.name
  }
}

l = arrayof(Foo)
l.append(Foo("foo1"))
println "got to 1"
l.append(Foo("foo2"))
println "got to 2"
