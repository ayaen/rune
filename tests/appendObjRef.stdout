got to 1
got to 2
